core/src/services/s3/lister.rs

```diff
@@ -133,3 +136,86 @@ impl oio::PageList for S3Lister {
         Ok(())
     }
 }
+
+// refer: https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectVersions.html
+pub struct S3ObjectVersionsLister {
```

Comment:
Perhaps we don't need a new list. I feel that most logic is similar.

Reply:
Add a method to `S3Lister`, such as `versioning_next_page()` ?  and like this ?
```
async fn next() {
    if (self.core.enable_versioning) {
       versioning_next_page()
       return; 
   }
  original_next()
}
``` 
I don't think this is a good idea, in a single `op.List()` call,  the`enable_versioning` flag cannot switch between true and false. 
Maybe we could add a function field in `S3Lister` struct, and decide which one to use based on  the `enable_versioning` flag ?

Reply:
I rechecked the API and found it's better to split it into two lists. Let's keep the current implementation.